x-pack/plugins/security_solution/public/timelines/components/side_panel/hooks/use_detail_panel.tsx

```diff
@@ -97,23 +109,50 @@ export const useDetailPanel = ({
 
   const openEventDetailsPanel = useCallback(
     (eventId?: string, onClose?: () => void) => {
-      if (eventId) {
+      if (isSecurityFlyoutEnabled) {
+        openFlyout({
+          right: {
+            id: DocumentDetailsRightPanelKey,
+            params: {
+              id: eventId,
+              indexName: eventDetailsIndex,
+              scopeId,
+            },
+          },
+        });
+        telemetry.reportDetailsFlyoutOpened({
+          location: scopeId,
+          panel: 'right',
+        });
+      } else if (eventId) {
         loadDetailsPanel({
           panelView: 'eventDetail',
           params: { eventId, indexName: eventDetailsIndex },
         });
+        onPanelClose.current = onClose ?? noopPanelClose;
       }
-      onPanelClose.current = onClose ?? noopPanelClose;
     },
-    [loadDetailsPanel, eventDetailsIndex]
+    [isSecurityFlyoutEnabled, openFlyout, eventDetailsIndex, scopeId, telemetry, loadDetailsPanel]
   );
 
   const openHostDetailsPanel = useCallback(
     (hostName: string, onClose?: () => void) => {
-      loadDetailsPanel({ panelView: 'hostDetail', params: { hostName } });
-      onPanelClose.current = onClose ?? noopPanelClose;
+      if (isSecurityFlyoutEnabled) {
```

Comment:
how do you trigger this flyout?

Reply:
I'm a bit ashamed to say but I do not know! I've only seen alerts and events within the session viewer so far.

Reply:
now I have one question: the hook exposes the `handleOnDetailsPanelClosed` function, but the only place we're using it is within the test. Should I just keep the code as is or try to get the test working in a different way? wdyt @michaelolo24 @christineweng?

Reply:
If you want to leave it as is and leave cleanup separately, I'm fine with that, but if you have time, I would remove it. iirc this hook was originally made to compartmentalize logic for the session view, then expanded for the initial/now gone alert details page.